streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -910,6 +900,11 @@ public void addRecords(final TopicPartition partition, final Iterable<ConsumerRe
         }
     }
 
+    @Override
+    public void addFetchedMetadata(final TopicPartition partition, final ConsumerRecords.Metadata metadata) {
```

Comment:
Wouldn't it be good to avoid taking on unrelated refactoring TODOs in this PR? It seems better to me to leave it to whoever decides to pick up that TODO and instead just keep this PR focused on the feature.